haystack/management/commands/update_index.py

```diff
@@ -85,6 +85,8 @@ def do_update(backend, index, qs, start, end, total, verbosity=1, commit=True):
     # in memory. Useful when reindexing large amounts of data.
     small_cache_qs = qs.all()
     current_qs = small_cache_qs[start:end]
+    
+    index.pre_process_data(current_qs)
```

Comment:
Interesting idea, could definitely do that but the purpose was not to alter the queryset for other people to use but rather do some things based on that query set.  I can see the additional use case, but that was not my focus and I hesitate to what to do that.